cmod/board.py

```diff
@@ -1,194 +1,297 @@
+   try:
+      self.type = jsonmap['type']
+      self.mode = int(jsonmap['mode'])
+      self.channel = int(jsonmap['channel'])
+      self.coordinates = {
+        "default": jsonmap['coordinates']['default'],
+        "calibrated": jsonmap['coordinates']['calibrated'] if len(jsonmap['coordinates']['calibrated']) > 0 else [] # a stack
+      }
+   except KeyError as e:
+     raise ValueError(e.msg)
+
+
+   # TODO: add the conditions calculated per detector
+   # self.vis_coord = {}
+   # self.vis_M = {}
+   # self.lumi_coord = {}
+   self.logger = board.cmd.devlog("Det")
+
+
+   # Additional parsing.
+   if (self.orig_coord[0] > gcoder.GCoder.max_x() or  #
+       self.orig_coord[1] > gcoder.GCoder.max_y() or  #
+       self.orig_coord[0] < 0 or self.orig_coord[1] < 0):
+     self.logger.warning(f"""
+       The specified detector position (x:{self.orig_coord[0]},
+       y:{self.orig_coord[1]}) is outside of the gantry boundaries
+       (0-{gcoder.GCoder.max_x()},0-{gcoder.GCoder.max_y()}). The expected
+       detector position will be not adjusted, but gantry motion might not
+       reach it. Which mean any results may be wrong.""")
+       # TODO: handle this in the gantry motions when cannot reach sipm
+
+
+ def __str__(self):
+   return str(self.__dict__)
+
+
+ def __dict__(self):
+   return {
+       'type': self.type,
+       'mode': self.mode,
+       'channel': self.channel,
+       'coordinates': self.coordinates,
+       # 'Luminosity coordinates': self.lumi_coord,
+       # 'Visual coordinates': self.vis_coord,
+       # 'FOV transformation': self.vis_M
+   }
+
+
 
 
 class Board(object):
-  """
-  Class for storing a board type an a list of det x-y positions
-  """
-  def __init__(self, cmd):
-    self.boardtype = ""
-    self.boarddescription = ""
-    self.boardid = ""
-    self.det_map = {}
-    self.cmd = cmd # Reference to main object
-    self.logger = self.cmd.devlog("Board")
-
-  def clear(self):
-    self.boardtype = ""
-    self.boarddescription = ""
-    self.boardid = ""
-    self.det_map = {}
-
-  def set_boardtype(self, file):
-    if any(self.dets()) or not self.empty():
-      self.logger.warning("""
-        The current session is not empty. Loading a new boardtype will erase any
-        existing configuration for the current session""")
-
-    jsonmap = json.loads(open(file, 'r').read())
-    self.boardtype = jsonmap['board type']
-    self.boarddescription = jsonmap['board description']
-    self.boardid = jsonmap['board id']
-
-    for detid in jsonmap['detectors']:
-      self.det_map[detid] = Detector(jsonmap['detectors'][detid], self)
-
-  def load_calib_file(self, file):
-    if not self.empty():
-      self.logger.warning("""
-       The current session is not empty. Loading a new boardtype will erase any
+ """
+ Class for storing a board type an a list of det x-y positions
+ """
+ def __init__(self, cmd):
+   self.filename = ""
+   self.type = ""
+   self.description = ""
+   self.id = UUID.uuid4()
```

Comment:
This needs to be changed. This will not run on an x86 machine.

Reply:
Do you know an alternative method that does work on x86 machine?

Reply:
This ID should be the unique string (number? not sure of the format right now) for the tileboard, not the control interface. So I think this should be a default (empty string or -1) value.

Reply:
did -1 for now